### PR TITLE
Add accfg benchmarks

### PR DESCRIPTION
**Pull request by JosseVanDelm**
Benchmarks for accfg

### CODE REVIEW

**Review by AntonLydike — APPROVED** [review]
:heart:

**Comment by AntonLydike** [comment]
@JosseVanDelm do you need any help with this?

File: kernels/streamer_matmul/transform_matmul.transform.mlir
Comment (JosseVanDelm) [inline kernels/streamer_matmul/transform_matmul.transform.mlir:1]: This change is unnecessary

File: kernels/streamer_matmul/genbenchmark.py [hunk 1]
```diff
@@ -1,30 +1,158 @@
...
+
+from xdsl.builder import ImplicitBuilder
+from xdsl.dialects import arith, builtin, func, linalg, transform
+from xdsl.ir import Block, Region
+from xdsl.printer import Printer
 
 from benchmark.snax_benchmark import SNAXBenchmark
 
-if __name__ == "__main__":
+i8 = builtin.IntegerType(builtin.IntAttr(8))
```
Comment (jorendumoulin) [inline kernels/streamer_matmul/genbenchmark.py:11]:
```suggestion
from xdsl.builtin import i8
```

File: kernels/streamer_matmul/genbenchmark.py [hunk 2]
```diff
@@ -1,30 +1,158 @@
...
+#
+#    binaries = {
+#        "run0": "matmul.x",
+#        "run1": "half_tiled_matmul.x",
+#        "run2": "transform_matmul.x",
+#        "run3": "dynamic_matmul.x",
+#    }
+#
+#    for folder, binary in binaries.items():
+#        run_all(binary, folder)
```
Comment (jorendumoulin) [inline kernels/streamer_matmul/genbenchmark.py:158]: comments

**Review by jorendumoulin — COMMENTED** [review]
Looks very nice!
But this does complicate one of the most important kernels.
Do you think it make sense to separate benchmarks (long-running tests that track performance) and kernels (fast running as-simple-as possible tests that track correctness)?
I thought the /benchmark folder was meant for this

**Comment by JosseVanDelm** [comment]
@jorendumoulin I tend to agree that separation is a good idea here.
I already think the current makefile is a bit impossible to follow.
I will separate accordingly, thanks!

**Comment by JosseVanDelm** [comment]
@jorendumoulin there is still some benchmarking code in kernels/tiled_matmul and kernels/tiled_add.
I suggest we remove this in a different pull request

**Comment by jorendumoulin** [comment]
> @jorendumoulin there is still some benchmarking code in kernels/tiled_matmul and kernels/tiled_add. I suggest we move this inside the new folder in a different pull request

yes, sounds good

**Pull request merged**